Circular_Doubly_Linked_List/Circular_Doubly_Linked_List.py

```diff
@@ -0,0 +1,302 @@
+        self.head = head
+
+    def pretty_print(self):
+
+        cir_dbl_list_str = ''
+
+        current_node = self.head
+
+        while current_node:
+            cir_dbl_list_str += '-({0:1d})-'.format(current_node.key)
+
+            current_node = current_node.next
+            if current_node == self.head:
+                break
+
+        return cir_dbl_list_str
+
+    # Insert node at the end of the circular doubly linked list
+    def append(self, key):
+        new_node = node(key)
+
+        # If there are no elements in the circular doubly linked list
+        if self.head is None:
+            self.head = new_node
+            new_node.next = self.head
+            new_node.prev = self.head
+
+        # If there are more elements in the circular doubly linked list
+        else:
+            last_node = self.head.prev
+
+            last_node.next = new_node
+            new_node.next = self.head
+
+            new_node.prev = last_node
+            self.head.prev = new_node
+
+    # Insert node at the beginning of the circular doubly linked list
+    def prepend(self, key):
+        new_node = node(key)
+
+        # If there are no elements in the circular doubly linked list
+        if self.head is None:
+            self.head = new_node
+            new_node.next = self.head
+            new_node.prev = self.head
+
+        else:
+
+            # If not an empty circular doubly linked list - the next and previous
+            # pointers for the head, new node and last node need to be updated
+
+            last_node = self.head.prev
+
+            # New node pointers updated
+            new_node.next = self.head
+            new_node.prev = last_node
+
+            # head , last node pointers updated
+            self.head.prev = new_node
+            last_node.next = new_node
+
+            # Update new head
+            self.head = new_node
+
+    # Insert after a specific node in the circular doubly linked list
+    def insert_after_node(self, afterkey, key):
+        current_node = self.head
+        while current_node:
+
+            if current_node.next == self.head and current_node.key == afterkey:
+                # New node is to be appended to the list
+                self.append(key)
+                return
+
+            # New node to be added in between nodes
+            elif current_node.key == afterkey:
+                new_node = node(key)
+                next_node = current_node.next
+
+                # Update next pointers
+                current_node.next = new_node
+                new_node.next = next_node
+
+                # Update prev pointers
+                new_node.prev = current_node
+                next_node.prev = new_node
+
+                return
+            else:
+                if current_node.next == self.head:
+                    break
+            current_node = current_node.next
+    # Insert before a specific node in the circular doubly linked list
```

Comment:
Add new line above comment instead of below it.

Reply:
Moved the new line above the comment.